packages/runtime/src/logicsig.ts

```diff
@@ -0,0 +1,173 @@
+import {
+  Account, decodeAddress, encodeAddress,
+  generateAccount, MultiSig, MultiSigAccount,
+  multisigAddress, signBytes, verifyBytes
+} from "algosdk";
+
+import { compareArray } from "../src/lib/compare";
+import { convertToString, stringToBytes } from "../src/lib/parsing";
+
+export class LogicSig {
+  logic: Uint8Array;
+  args: Uint8Array[];
+  sig: Uint8Array;
+  msig: MultiSig | undefined;
+  SignatureAddress: string;
+
+  constructor (program: string, args: Uint8Array[]) {
+    this.logic = stringToBytes(program);
+    this.args = args;
+    this.sig = new Uint8Array(0);
+    this.msig = undefined;
+    this.SignatureAddress = generateAccount().addr;
+  }
+
+  /**
+   * Creates signature (if no msig provided) or multi signature otherwise
+   * @param secretKey sender's secret key
+   */
+  sign (account: Account, msig?: MultiSigAccount): void {
```

Comment:
Can you please explain as why we're passing account here? When using SDK we usually pass a sk (eg. `lsig.sign(delegator.sk)` )

Reply:
here we need a function to retrieve public key from secret key, i though that will create extra dependency, but i think sk is good, updating

Reply:
@ratik21 do you know library which has 64 length secret key and public key or how to change it ? i could find tweetnacl , it has 32 length.

Reply:
got it, nvm